### DISCUSSION TITLE
How can I detect screen navigation on a POS system?

### DISCUSSION THREAD

**Discussion by paryken** [microsoft/Dynamics365Commerce.Solutions#277] [Q&A]
I want to detect screen transitions and trigger a function when the screen changes. I've already tried using the dispose method, but it takes too long to activate after the screen has transitioned. I would like to trigger the function at the moment of the screen transition or just before it. Does anyone know how to do this?

My specific scenario involves a screen that extends PaymentView.PaymentViewExtensionCommandBase, particularly when it returns to the HomeView. How can I handle this effectively?

**Comment by zhangguanghuib** [comment]
Can you please check this code:  https://github.com/zhangguanghuib/NewCommerceSDK/blob/main/POS_Samples/Solutions/SampleExtensions/Pos/Extend/ViewExtensions/Payment/PaymentViewCommandV2.ts
 The key point is:

<img width="930" alt="image" src="https://github.com/microsoft/Dynamics365Commerce.Solutions/assets/14832260/ca7675c8-06e2-4073-b2f0-afbed2b59e58">

Call this code will get the current view name:

 let viewName: string = Commerce.ViewModelAdapter.getCurrentViewName();

In case the current view name is not the PaymentView,   customer can call HWS API to withdraw money from the cash change machine.

Please let me know if you have any other question
